src/software/ai/hl/stp/play/play_fsm.h

```diff
@@ -9,15 +10,28 @@ using PriorityTacticVector      = std::vector<TacticVector>;
 using ConstTacticVector         = std::vector<std::shared_ptr<const Tactic>>;
 using ConstPriorityTacticVector = std::vector<ConstTacticVector>;
 
+// Struct used to communicate between plays
+struct InterPlayCommunication
+{
+    std::optional<PassWithRating> last_committed_pass;
```

Comment:
will every play be expected to reset `last_committed_pass` or will `last_committed_pass` be available for multiple play transitions? If so then will we need to add logic somewhere that checks if this information is still valid?

Reply:
my thought was that the consumer of `last_committed_pass` would reset it, but STP could also do it. I don't think I can predict what the best place to reset it would be.

Reply:
a problem that I can see with the consumer of last_committed_pass resetting it is: what if the pass happens and then we transition into an unexpected play (StopPlay, HaltPlay), then the old pass information is stale and shouldn't be used. maybe this message should be reset by playselectionfsm

Reply:
As the architecture is currently set up, it would have to be reset in the stopplay/haltplay itself. How does that sound?

Reply:
that sounds like a good solution to me

Reply:
cant the same logic be applied to any play? if we set a pass then transition to defense then offense, the pass might be stale

Reply:
yeah so if we were to use a defense play then it'd have to reset it too